src/makeCalculator.js

```diff
@@ -37,7 +37,49 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  return {
+    result: 0,
+
+    add: function(number) {
+      this.result += number;
+
+      return this;
+    },
+
+    subtract: function(number) {
+      this.result -= number;
+
+      return this;
+    },
+
+    multiply: function(number) {
+      this.result *= number;
+
+      return this;
+    },
+
+    divide: function(number) {
+      if (number !== 0) {
+        this.result /= number;
+      }
+
+      return this;
+    },
+
+    reset: function() {
+      this.result = 0;
+
+      return this;
+    },
+
+    operate: function(callback, number) {
+      if (typeof callback === 'function') {
+        callback.call(this, number);
+      }
```

Comment:
I don't see any sense for that condition, because callback should be every time with type of 'function'